keel-api/src/main/kotlin/com/netflix/spinnaker/keel/rest/ResourceController.kt

```diff
@@ -94,9 +94,9 @@ class ResourceController(
     log.error(e.message)
   }
 
-  @ExceptionHandler(HttpMessageNotReadableException::class)
+  @ExceptionHandler(HttpMessageConversionException::class)
```

Comment:
This is just a broader exception. If you have validation rules in the `init` block of the data class it throws the more general exception.